Const-qualify member functions for newer compilers

Ensure that some member functions are callable on const-qualified
objects with newer system compilers, as they will otherwise refuse to
compile the source.

Author: Lars Viklund <zao@zao.se>
diff -ru ncurses-5.9.orig/c++/cursesf.h ncurses-5.9/c++/cursesf.h
--- ncurses-5.9.orig/c++/cursesf.h	2005-08-13 20:08:24.000000000 +0200
+++ ncurses-5.9/c++/cursesf.h	2020-04-02 16:03:10.219252152 +0200
@@ -381,7 +381,7 @@
     uptr->m_user = user;
   }
 
-  inline void *get_user() {
+  inline void *get_user() const {
     UserHook* uptr = reinterpret_cast<UserHook*>(::form_userptr (form));
     assert (uptr != 0 && uptr->m_back==this && uptr->m_owner==form);
     return uptr->m_user;
diff -ru ncurses-5.9.orig/c++/cursesm.h ncurses-5.9/c++/cursesm.h
--- ncurses-5.9.orig/c++/cursesm.h	2005-08-13 20:10:36.000000000 +0200
+++ ncurses-5.9/c++/cursesm.h	2020-04-02 16:03:21.407103674 +0200
@@ -242,7 +242,7 @@
     uptr->m_user = user;
   }
 
-  inline void *get_user() {
+  inline void *get_user() const {
     UserHook* uptr = STATIC_CAST(UserHook*)(::menu_userptr (menu));
     assert (uptr != 0 && uptr->m_back==this && uptr->m_owner==menu);
     return uptr->m_user;
diff -ru ncurses-5.9.orig/c++/cursesp.h ncurses-5.9/c++/cursesp.h
--- ncurses-5.9.orig/c++/cursesp.h	2008-08-16 19:20:23.000000000 +0200
+++ ncurses-5.9/c++/cursesp.h	2020-04-02 16:03:29.522995966 +0200
@@ -58,7 +58,7 @@
     const PANEL*        m_owner;     // the panel itself
   } UserHook;
 
-  inline UserHook *UserPointer()
+  inline UserHook *UserPointer() const
   {
     UserHook* uptr = reinterpret_cast<UserHook*>(
                            const_cast<void *>(::panel_userptr (p)));
@@ -77,7 +77,7 @@
   }
   // Set the user pointer of the panel.
 
-  void *get_user()
+  void *get_user() const
   {
     UserHook* uptr = UserPointer();
     void *result = 0;
